zerver/views/invite.py

```diff
@@ -232,6 +233,67 @@ def resend_user_invite_email(
     return json_success(request)
 
 
+@require_member_or_admin
+@typed_endpoint
+def edit_multiuse_invite(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    invite_id: PathOnly[int],
+    stream_ids: Json[list[int]] | None = None,
+    invite_as: Annotated[
+        Json[int] | None,
+        check_int_in_validator(list(PreregistrationUser.INVITE_AS.values())),
+    ] = None,
+    include_realm_default_subscriptions: Json[bool] = False,
+) -> HttpResponse:
+    try:
+        invite = MultiuseInvite.objects.get(id=invite_id)
+    except MultiuseInvite.DoesNotExist:
+        raise JsonableError(
+            _("Invite does not exist with id: {invite_id}.").format(invite_id=invite_id)
+        )
+
+    if (
+        invite.referred_by_id != user_profile.id
+        and not user_profile.is_realm_admin
+        and not user_profile.can_create_multiuse_invite_to_realm()
+    ):
+        # Guest users case will not be handled here as it will
+        # be handled by the decorator above.
+        raise JsonableError(_("Insufficient permission"))
+
+    if invite_as is not None:
+        require_admin = invite_as in [
+            PreregistrationUser.INVITE_AS["REALM_OWNER"],
+            PreregistrationUser.INVITE_AS["REALM_ADMIN"],
+            PreregistrationUser.INVITE_AS["MODERATOR"],
+        ]
+        check_role_based_permissions(invite.invited_as, user_profile, require_admin=require_admin)
+        invite.invited_as = invite_as
+
+    streams: list[Stream] = []
+    if stream_ids is not None:
+        for stream_id in stream_ids:
+            try:
+                (stream, sub) = access_stream_by_id(user_profile, stream_id)
+            except JsonableError:
+                raise JsonableError(
+                    _("Stream does not exist with id: {stream_id}.").format(stream_id=stream_id)
+                )
+            streams.append(stream)
+
+        if streams and not user_profile.can_subscribe_others_to_all_accessible_streams():
+            raise JsonableError(
+                _("You do not have permission to subscribe other users to channels.")
```

Comment:
`access_streams_for_invite` can be used so that the permission check is done correctly. Currently, this does not check stream-level settings.